[Tools] Adds assignment placer to place files in pre-existing submissions

andrewaikens87 opened the pull request:
This PR takes a collection of files and submits them into a current gradeable. These files can then be used in the plagiarism module via regex. It may be a good idea to generate a unique string to make regex easier.

andrewaikens87 committed:
Tools

andrewaikens87 committed:
fixed

andrewaikens87 committed:
assign Placer tweak

andrewaikens87 committed:
fix install.sh

MasterOdin commented on install_lichen.sh:
```diff
@@ -30,6 +30,7 @@ fi
 # compile & install the tools
 
 mkdir -p ${lichen_installation_dir}/bin
+mkdir -p ${lichen_installation_dir}/tools
```
Let's also make the `assignments` folder here, given that it's required.

MasterOdin commented on tools/assignmentPlacer.py:
```diff
@@ -0,0 +1,28 @@
+import os, sys
```
can we rename this to `assignment_placer.py` to better follow how other files are named.

MasterOdin commented on tools/assignmentPlacer.py:
```diff
@@ -0,0 +1,28 @@
+import os, sys
+import shutil
+
+gradeable_path = '/var/local/submitty/courses/s20/sample/submissions/grades_released_homework_autohiddenEC'
```
It would be good to have this be customizable via args. I think for a v1, we can probably just expect the user to provide the full path they want to copy to.

MasterOdin commented on tools/assignmentPlacer.py:
```diff
@@ -0,0 +1,28 @@
+gradeable_path = '/var/local/submitty/courses/s20/sample/submissions/grades_released_homework_autohiddenEC'
+
+if __name__ == "__main__":
+    if not os.path.isdir('assignments') or not os.path.isdir(gradeable_path):
```
Let's use an absolute path to this folder instead of it being relative (assuming we expect to always pick up out of `${lichen_installation_dir}/tools/assignments`.

Can do something like:
```
import pathlib
assignment_path = str(pathlib.Path(__file__).parent.absolute() / 'assignments')
```

MasterOdin commented on tools/assignmentPlacer.py:
```diff
@@ -0,0 +1,28 @@
+    for root, dirs, files in os.walk(gradeable_path):
+        for name in dirs:
+            print(f'{root} {name}')
+            if (root.count('/') == gradeable_path_count):
```
Trying to understand this check, is this to limit us to only searching the top-level of `gradeable_path` and no deeper? Are we actually walking through the entirety of `gradeable_path` and doing something meaningful with those layers?

andrewaikens87 replied:
We are only walking into one child from the root (i.e. `gradeable_path/user_id` to find the highest submission number), I can change this to not use os.walk but rather a combination of os.listdirs since the depth we use is fixed for each execution.

MasterOdin commented on tools/assignmentPlacer.py:
```diff
@@ -0,0 +1,28 @@
+if __name__ == "__main__":
+    if not os.path.isdir('assignments') or not os.path.isdir(gradeable_path):
+        print('The assignments directory or the gradeable does not exist. Please create these.')
+        sys.exit(1)
```
`raise SystemExit('The assignments directory or the gradeable does not exist. Please create those.')`

MasterOdin commented on tools/assignmentPlacer.py:
```diff
@@ -0,0 +1,28 @@
+                ci+=1
+                placed += 1
+            if(placed == num_to_place):
+                sys.exit()
```
`raise SystemExit`

MasterOdin reviewed (changes requested):
Prefer to use `SystemExit` exception whenever possible over `sys.exit` unless you want specific error codes as it means we do not have to `import sys` then.

andrewaikens87 committed:
update assignment_placer

andrewaikens87 committed:
tweaks

andrewaikens87 committed:
small twekas

andrewaikens87 committed:
fix hardcode semester

andrewaikens87 commented:
@MasterOdin I have updated the script to reflect your requested changes. I have included args for the semester, gradeable, and course. In addition to this, as I mentioned in in the original post, I have added a unique hash for the files to make it easier to regex when setting up the lichen configuration. Please let me know if there is anything I missed!

MasterOdin committed:
Update assignment_placer.py
